File: backend/src/main/java/br/marcoswolf/pulsedelivery/model/DeliveryPerson.java
package br.marcoswolf.pulsedelivery.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "delivery_person")
public class DeliveryPerson extends User {
    private String phone;
    private String document;

    @Embedded
    private Image image;
}

File: frontend/src/modules/deliveryPerson/types/DeliveryPerson.ts
import type { User } from "../../user/types/User";
import type { Address } from "../../../shared/types/Address";
import type { Image } from "../../../shared/types/Image";

export type DeliveryPerson = {
    user: User;
    document: string;
    phone: string;
    address: Address;
    image: Image;
};